### PR TITLE
Minor Spider Rebalance

### PR DESCRIPTION
**Pull request by DrAmazing343**
## About The Pull Request
Breacher and Viper player-controlled Spiders have swapped castes; Vipers now come from Abnormal Eggs, while Breachers will now come from Enriched Eggs. This means that Vipers may be seen more, while Breachers require the Nest to acquire at least a single kill to wrap and create an Enriched Egg from. Nurse Spiders also perform their heal action much faster, at the cost of a bit less healing. It was very difficult to heal others due to the long do_after and thus we saw very few Nurse Spiders.

Breacher Spiders also are now NOT immune to cold, and will die if they try to space a room or go into space.

## Why It's Good For The Game
Breacher Spiders are in kind of a difficult spot, both in theming and balance; their space resistance made them able to vent the station in multiple places by breaking key walls, and they had no need to commit to their sieges because they could escape into space at all times. It also broke part of the dynamic of Spiders wanting to keep the atmosphere in the station, and disincentivizing Breachers from venting the station also makes them more coherently a part of the Nest because they won't reduce the Nest's habitable area.

Viper Spiders were seen quite little instead of the other options in the Enriched Egg; now, hopefully we'll see more. Likewise, the Breacher is quite powerful for being an Abnormal Egg, so perhaps now we'll see more focus on cocooning to unlock the powerful Breachers, or just overall less of them, which will help even the odds.

Nurse Spiders were VERY underutilized despite their power— the do_after for their healing just felt obnoxiously long, and especially in tense situations, you'd have shuffle and Spiders yelling at one another because two of them have to Space Hulk the hallway for a semi-extended period of time. Now, Nurses will be able to heal and move on a bit faster. I don't believe this will create too many issues, balance-wise, as the time-to-kill on Spiders is still quite low for all but the tankiest against lasers and even welders.
## Changelog
:cl: Jane
balance: Spider Breachers and Vipers have swapped castes. Breachers come from Enriched Eggs while Vipers come from Abnormals! Breachers can no longer survive in spaced rooms/temperatures. The Nurse wraps wounds faster, but with less damage healed per wrap.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/basic/space_fauna/spider/giant_spider/giant_spiders.dm [hunk 1]
```diff
@@ -317,7 +318,7 @@
 	melee_damage_lower = 5
 	melee_damage_upper = 10
 	unsuitable_atmos_damage = 0
-	minimum_survivable_temperature = 0
+	minimum_survivable_temperature = 75
```
Comment (CRITAWAKETS) [inline code/modules/mob/living/basic/space_fauna/spider/giant_spider/giant_spiders.dm:321]: Undocumented change 👀

**Comment by Loafin34** [comment]
reduce the nurse's healing but keep the speed so that it's more gradual.

**Comment by DrAmazing343** [comment]
> reduce the nurse's healing but keep the speed so that it's more gradual.

Done and done, this is something I should've considered more but that's why GitHub is public. Thanks!

**Comment by Xkallubar** [comment]
Removing breacher spiders being space immune defeats their purpose. Now the only thing special about them is going to be their ability to break non reinforced walls.
Delete breachers and give the wall breaking to tank spiders if this has to happen. Maybe make broodmothers a bit stronger aswell.

**Comment by DrAmazing343** [comment]
> Removing breacher spiders being space immune defeats their purpose. Now the only thing special about them is going to be their ability to break non reinforced walls. Delete breachers and give the wall breaking to tank spiders if this has to happen. Maybe make broodmothers a bit stronger aswell.

I disagree that their purpose is exclusively to go breach into space and become a pest; they still hold a niche within spider society as point spiders getting the Nest out of tricky spots without resorting to spacing the area. I will also not be touching broodmother stats as that's out of the scope of this PR and my anecdotal experience.

**Comment by Sadboysuss** [comment]
> Removing breacher spiders being space immune defeats their purpose. Now the only thing special about them is going to be their ability to break non reinforced walls. Delete breachers and give the wall breaking to tank spiders if this has to happen. Maybe make broodmothers a bit stronger aswell.

breacher spiders can break r-walls

**Comment by EricZilla** [comment]
I feel like if we were gonna have a spaceproof spider it should walk slow and spin web fast, so it's goal is protection of atmos for the nest

**Comment by SomeoneYouProbablyKnow** [comment]
> Removing breacher spiders being space immune defeats their purpose. Now the only thing special about them is going to be their ability to break non reinforced walls.
Delete breachers and give the wall breaking to tank spiders if this has to happen. Maybe make broodmothers a bit stronger aswell.

Yea, honestly even making the broodmother space immune or space resistant might not be that bad as long as the eggs need atmos to survive. We have way too many specialized types so half the time the hive can't do the things it needs to do. And not like the crew needs to change their behavior against spider most spider types.

As for the venomy boy, i hate them on principle because they rely on their speed and high speed combat is absolute ass and it makes it really difficult to run away from spiders, be geared enough to survive or die. 
Leaving them as the only combat focused type in abnormal eggs will make it so people pick it over other support types. 

Honestly nerfing them and making them a normal type (instead of hunters, same function anyway) might work but that's an even fiddlier balance situation.

**Comment by DaCoolBoss** [comment]
Full space immune will definitely be abused to create infinite spider slogfests with broodmothers hiding in space between egglaying sessions. Spider atmos control should be weak enough to keep spiders on station. Maybe take inspiration from diving bell spiders?

**Comment by ElGitificador** [comment]
just think that spiders dont need any nerfs, they re usually stomped or ignored for what i ve seen

**Comment by DrAmazing343** [comment]
This nerf is specifically to the brand of spider that can leave the Nest entirely and go cause problems stationwide for little commitment and no real advancement to the Nest, actively creating non-habitable environs for the other spiders. It's also a breed that is terrifyingly powerful even post-nerf, able to get spider colonies out of unwinnable backed-against-the-wall situations in maint, and that's why they require at least a kill to spawn one now.

> As for the venomy boy, i hate them on principle because they rely on their speed and high speed combat is absolute ass and it makes it really difficult to run away from spiders, be geared enough to survive or die.
Leaving them as the only combat focused type in abnormal eggs will make it so people pick it over other support types.

I can understand this, but spiders aren't necessarily a threat that a non-geared crewmate should be able to fight. The point about not being able to run away is a better one, however, a spider that runs too far from the Nest often dies. Either way, we hardly see Vipers, so I'm quite interested to see if they WOULD become part of the "spider meta." If Vipers are indeed too overpowering after this change, I'd be willing to go back and tick the values or do some minor rework to make them more fair.

**Comment by SomeoneYouProbablyKnow** [comment]
>Either way, we hardly see Vipers, so I'm quite interested to see if they WOULD become part of the "spider meta." 

Well part of it is that they come from the same spawn as broodmother and tarantula, there's A LOT of competition there. And their individual power level is that of an elite spider make no mistake. Which is why i'd rather see them preemptively nerfed and put into common spider tier. 

> If Vipers are indeed too overpowering after this change, I'd be willing to go back and tick the values or do some minor rework to make them more fair.

Fair nuff, spiders are in a weird place right now, maybe bunch of vipers is what they need.

**Comment by SethLafuente** [comment]
Specifically for the breached spider changes I don't quite agree with removing their space resistant, I don't think breached spiders would do what you describe unless they came from xenobiology or the entire spider nest died it's not within their interest to cause station breached because that screws over other spiders and killing your own team antags is against the rules. This is why I didn't like breacher spider spawning from gold slime extract.

I would like to see some testing of this when I'm able too and if it seems like a good change I won't do anything about it I guess is good to swap things around and see how they impact the gameplay.

Breacher is a specialized spider its purpose is to breach in or modify station walls to fit the hives desire to be able to close up breaches requires the ability to not die by those breaches.

Swapping viper with breacher I find particularly odd abnormal eggs are meant to be specialized spider types while enriched eggs are solid strong core spiders breacher does not fit that denomination also enriched eggs are just better version of base spiders so thematically it destroys that.

Viper has been neglected I think it would be fit to move it back to enriched clusters once is buffed in some way it's meant to be a stronger core spider version of the hunter spider that lasts but it dosent last, I'll see what I should do about that.

Any change to spiders is good on the lung run since it keeps the conversation of the antag alive so this is good even if the changes might not be so good.

**Pull request merged**